rest/replicator_test.go

```diff
@@ -3810,3 +3810,236 @@ func TestActiveReplicatorPullConflictReadWriteIntlProps(t *testing.T) {
 		})
 	}
 }
+
+// This test ensures that the tombstone revision wins over non-tombstone revision
+// whilst applying default conflict resolution policy through pushAndPull replication.
+func TestDefaultConflictResolverWithTombstone(t *testing.T) {
+	if base.GTestBucketPool.NumUsableBuckets() < 2 {
+		t.Skipf("test requires at least 2 usable test buckets")
+	}
+	if !base.TestUseXattrs() {
+		t.Skip("This test only works with XATTRS enabled")
+	}
+	defer base.SetUpTestLogging(base.LevelDebug, base.KeyAll)()
+
+	// startReplication starts the replication tasks and asserts that the start up
+	// operation is successful. It returns a teardown function to stop the replication
+	// that can be called in defer.
+	startReplication := func(ar *db.ActiveReplicator) func() {
+		require.NoError(t, ar.Start(), "Error starting replication")
+		return func() { require.NoError(t, ar.Stop(), "Error stopping replication") }
+	}
+
+	// requireBodyEmpty asserts that the specified document body is empty.
+	requireBodyEmpty := func(rt *RestTester, docID string) {
+		var body []byte
+		_, err := rt.Bucket().Get(docID, &body)
+		require.EqualError(t, err, fmt.Sprintf("Error during Get %s: key not found", docID))
+	}
+
+	// waitForRevision waits until the specified tombstone revision is available
+	// in the bucket backed by the specified RestTester instance.
+	waitForTombstone := func(rt *RestTester, docID, revID string) {
+		require.NoError(t, rt.WaitForCondition(func() bool {
+			doc, _ := rt.GetDatabase().GetDocument(docID, db.DocUnmarshalAll)
+			return doc.IsDeleted()
+		}))
+		doc, err := rt.GetDatabase().GetDocument(docID, db.DocUnmarshalAll)
+		require.NoError(t, err, "Error reading document from bucket")
+		require.Equal(t, revID, doc.SyncData.CurrentRev)
+		require.Empty(t, doc.Body(), "Document body should be empty")
+	}
+
+	// requireBodyEmpty asserts that the specified document body is not empty.
+	requireBodyNotEmpty := func(rt *RestTester, docID, revID, bodyValue string) {
```

Comment:
More generally, I don't think 'requireBodyNotEmpty' is needed in this test (see comments below).